src/main/java/com/delphix/masking/initializer/maskingApi/BackupDriver.java

```diff
@@ -239,6 +247,32 @@ private void finishBackup() {
         }
     }
 
+    private boolean isBackupRequired(int inputMajorVersion, int inputMinorVersion, int inputMicroVersion) {
+        int majorVersion = 0;
+        int minorVersion = 0;
+        int microVersion = 0;
+        GetSystemInformations getSystemInformations = new GetSystemInformations();
+        apiCallDriver.makeGetCall(getSystemInformations);
```

Comment:
Nit: Every time this function is called, we will be making an API call to the engine to fetch the exact same data. Instead, we should cache this value as it will never change for a given run.

Reply:
Done.